### ISSUE TITLE
Why EF is disponsing Context and Closing/Opening new connections in a Single Command

### ISSUE THREAD

**Issue by danielscatigno-ncpc** [dotnet/efcore#26850]
Hi I'm having an issue that is only occurring in my local Windows 10 Sql Server Express Instance
It's damm slow while issuing some Commands

I have tested using an Windows Server as Database Server and the issues doesn't happen

So I checked the logs (Attached)
The log attached is only about a single command issued to the server

```C#
var postoQuery = UOW.GetRepository<PostoTrabalhoRepository>().GetBy(x => x.IdPlanta == registro.Id)
        .Include(x=>x.CentroCusto).Include(x=>x.UsuariosPermitidos);
         
registro.PostosTrabalho = postoQuery.ToList();
```

"GetBy" here is a IQueryable of PostoTrabalho

so my question is Why EF is disponsing Context and Closing/Opening new connections in a Single Command ?
![image](https://user-images.githubusercontent.com/85890187/143923070-a5007b8a-d4f0-4718-9c2f-3d9a629bf0c6.png)

Wasn't it supposed to use only one connection?
I suspect this is causing the problem as Windows 10 is limited to 20 concurrent connection
Something is blocking EntityFramework from creating a new connection

EF Core version: 5.0.12
Database provider: Microsoft.EntityFrameworkCore.SqlServer
Target framework: .NET 5.0
Operating system: Windows 10 Pro
IDE: Visual Code

[LogEntityMultipleConnections.txt](https://github.com/dotnet/efcore/files/7620349/LogEntityMultipleConnections.txt)

**Comment by ajcvickers** [comment]
@danielscatigno-ncpc There is not enough here to understand what is going on in the application. Please attach a small, runnable project or post a small, runnable code listing that reproduces what you are seeing so that we can investigate.

**Comment by danielscatigno-ncpc** [comment]
Sory for the Delay

When I was trying to build an example, I figure it out the log is a mix of various operations issued by the DbContext, but they may not be related to the command from which the Trouble is happening.

I'll Close this Issue as I found out the problem is related to some bug within Sql Server or Windows.
Sql Server is aleatory delaying the query , hitting 29 seconds or more, and sometimes it returns the results in less than a second.

I also found the option AsSplitQuery very useful, as it suppress this behavior of my Sql Server instance by splitting the problematic query into multiple ones
